src/seeds.quests.cpp

```diff
@@ -0,0 +1,1554 @@
+  hypha::ContentWrapper proposal_v_cw = proposal_v_doc.getContentWrapper();
+
+  check_type(proposal_doc, graph::PROPOSAL);
+
+  name proposal_stage = proposal_v_cw.getOrFail(VARIABLE_DETAILS, STAGE) -> getAs<name>();
+  name proposal_type = proposal_cw.getOrFail(FIXED_DETAILS, PROPOSAL_TYPE) -> getAs<name>();
+  checksum256 node_hash = proposal_cw.getOrFail(IDENTIFIER_DETAILS, NODE_HASH) -> getAs<checksum256>();
+  hypha::Document node_doc(get_self(), node_hash);
+
+  check(proposal_stage == proposal_stage_active, "can not vote for this proposal, it is not active");
+  check(!edge_exists(proposal_hash, voter), "quests: only one vote");
+
+  hypha::Document account_info_doc = get_account_info(voter, true);
+
+  if (proposal_type == proposal_type_quest) {
+    hypha::Edge::getOrNew(get_self(), get_self(), node_doc.getHash(), account_info_doc.getHash(), graph::VALIDATE);
+  } else {
+    hypha::ContentWrapper node_cw = node_doc.getContentWrapper();
+    checksum256 quest_hash = node_cw.getOrFail(IDENTIFIER_DETAILS, QUEST_HASH) -> getAs<checksum256>();
+    check(hypha::Edge::exists(get_self(), quest_hash, account_info_doc.getHash(), graph::VALIDATE), 
+      "quests: user can not vote for this proposal as it is not a validator");
+  }
+
+  action(
+    permission_level(contracts::proposals, "active"_n),
+    contracts::proposals,
+    "questvote"_n,
+    std::make_tuple(voter, amount, true, contracts::proposals)
+  ).send();
+
+  string label;
+  int64_t new_value;
+  int64_t favour_amount = 0;
+  int64_t against_amount = 0;
+
+  if (option == trust) {
+    label = FAVOUR;
+    new_value = amount + proposal_v_cw.getOrFail(VARIABLE_DETAILS, FAVOUR) -> getAs<int64_t>();
+    favour_amount = amount;
+  } else if (option == distrust) {
+    label = AGAINST;
+    new_value = amount + proposal_v_cw.getOrFail(VARIABLE_DETAILS, AGAINST) -> getAs<int64_t>();
+    against_amount = amount;
+  }
+
+  int64_t total_votes = proposal_v_cw.getOrFail(VARIABLE_DETAILS, TOTAL_VOTES) -> getAs<int64_t>();
+
+  update_node(&proposal_v_doc, VARIABLE_DETAILS, {
+    hypha::Content(label, new_value),
+    hypha::Content(TOTAL_VOTES, total_votes + 1)
+  });
+
+  hypha::ContentGroups vote_cgs {
+    hypha::ContentGroup {
+      hypha::Content(hypha::CONTENT_GROUP_LABEL, FIXED_DETAILS),
+      hypha::Content(TYPE, graph::VOTE),
+      hypha::Content(FAVOUR, favour_amount),
+      hypha::Content(AGAINST, against_amount)
+    },
+    hypha::ContentGroup {
+      hypha::Content(hypha::CONTENT_GROUP_LABEL, IDENTIFIER_DETAILS),
+      hypha::Content(PROPOSAL_HASH, proposal_hash),
+      hypha::Content(OWNER, voter)
+    }
+  };
+
+  hypha::Document vote_doc(get_self(), voter, std::move(vote_cgs));
+
+  hypha::Edge::write(get_self(), voter, proposal_hash, vote_doc.getHash(), voter);
+  hypha::Edge::write(get_self(), voter, proposal_hash, vote_doc.getHash(), graph::VOTED_BY);
+  // hypha::Edge::write(get_self(), voter, vote_doc.getHash(), proposal_hash, graph::VOTE_FOR);
```

Comment:
Oh well, I didn't include any of vote delegation. The edges are there so one could query who voted for the quest, and there is an edge with the name of the voter because it is faster to locate inside the contract

Reply:
Ok, so no delegation for now, fine.